src/gui/InteractiveModelView.cpp

```diff
@@ -0,0 +1,335 @@
+					break;
+				}
+			}
+		}
+	}
+	else
+	{
+		// find the lowest `ModelShortcut::times`
+		for (size_t i = 0; i < shortcuts.size(); i++)
+		{
+			if (doesShortcutMatch(&shortcuts[i], event))
+			{
+				// selecting the shortcut with the lowest `ModelShortcut::times`
+				if (found == false || minMaxTimes > shortcuts[i].times)
+				{
+					foundIndex = i;
+					minMaxTimes = shortcuts[i].times;
+				}
+				m_lastShortcut = shortcuts[i];
+				m_lastShortcutCounter = 1;
+				found = true;
+			}
+		}
+	}
+	if (found)
+	{
+		QString message = shortcuts[foundIndex].shortcutDescription;
+		showMessage(message);
+		processShortcutPressed(foundIndex, event);
+
+		event->accept();
+	}
+	else
+	{
+		// reset focus
+		if (event->key() != Qt::Key_Control
+			&& event->key() != Qt::Key_Shift
+			&& event->key() != Qt::Key_Alt
+			&& event->key() != Qt::Key_AltGr)
+		{
+			getGUI()->mainWindow()->setFocusedInteractiveModel(nullptr);
+		}
+	}
+	return found;
+}
+
+void InteractiveModelView::keyPressEvent(QKeyEvent* event)
+{
+	// this will run `HandleKeyPress()` for the widget that is focused inside MainWindow
+	getGUI()->mainWindow()->focusedInteractiveModelHandleKeyPress(event);
+}
+
+void InteractiveModelView::enterEvent(QEvent* event)
+{
+	m_lastShortcutCounter = 0;
+	m_lastShortcut.reset();
+
+	QString message = getShortcutMessage();
+	showMessage(message);
+
+	if (isVisible())
+	{
+		// focus on this widget so keyPressEvent works
+		getGUI()->mainWindow()->setFocusedInteractiveModel(this);
+	}
+}
+
+void InteractiveModelView::leaveEvent(QEvent* event)
+{
+	hideMessage();
+}
+
+bool InteractiveModelView::processPaste(const QMimeData* mimeData)
+{
+	if (Clipboard::hasFormat(Clipboard::MimeType::StringPair) == false) { return false; }
+
+	Clipboard::DataType type = Clipboard::decodeKey(mimeData);
+	QString value = Clipboard::decodeValue(mimeData);
+	bool shouldAccept = processPasteImplementation(type, value);
+	if (shouldAccept)
+	{
+		InteractiveModelView::stopHighlighting();
+	}
+	return shouldAccept;
+}
+
+void InteractiveModelView::overrideSetIsHighlighted(bool isHighlighted, bool shouldOverrideUpdate)
+{
+	setIsHighlighted(isHighlighted, shouldOverrideUpdate);
+}
+
+void InteractiveModelView::drawAutoHighlight(QPainter* painter)
+{
+	if (getIsHighlighted())
+	{
+		QColor fillColor = *s_usedHighlightColor;
+		fillColor.setAlpha(70);
+		painter->fillRect(QRect(1, 1, width() - 2, height() - 2), fillColor);
+
+		painter->setPen(QPen(*s_usedHighlightColor, 2));
+		painter->drawLine(1, 1, 5, 1);
+		painter->drawLine(1, 1, 1, 5);
+		painter->drawLine(width() - 2, height() - 2, width() - 6, height() - 2);
+		painter->drawLine(width() - 2, height() - 2, width() - 2, height() - 6);
+	}
+}
+
+QString InteractiveModelView::buildShortcutMessage()
+{
+	QString message = "";
+	std::vector<ModelShortcut> shortcuts(getShortcuts());
+	for (size_t i = 0; i < shortcuts.size(); i++)
+	{
+		message = message + QString("\"") + QKeySequence(shortcuts[i].modifier).toString()
```

Comment:
Please provide a switch/case to capture the correct remappings for Apple keys.  e.g. 

https://github.com/LMMS/lmms/blob/167b55a0b2773294550f4c6cfae3b8a3dd841aec/include/KeyboardShortcuts.h#L43-L48